### PR TITLE
refactor/fix/test(sqs): pool liquidity pricing refactor & clean up

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

The goal of this work is to refactor and clean up pool liquidity pricing logic, making the abstraction consistently denominated in the single unit (USDC, descaled with precision). A related [PR](https://github.com/osmosis-labs/sqs/pull/241) exists in sqs

This is also a pre-requisite towards the dynamic min liquidity pricing in SQS and pool liquidity pricing:
https://github.com/osmosis-labs/sqs/pull/224

As a side benefit, the following issue was resolved, making the pool liquidity capitalization be consistently in a single denomination
https://linear.app/osmosis/issue/DATA-48/understand-why-pool-tvl-has-become-scaled-updatetest-new-behavior

Additionally, the dependency on the asset list was removed from the node for no need.

Lastly, clean-ups of the `convertPool` function were achieved by encapsulating complex logic into separate functions for greater testability.

## Testing and Verifying

- Unit tests
- Will test in stage after backporting to `v25.x` 

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: go.mod
Comment (p0mvn) [inline go.mod:402]: Note to reviewer: I will retag this before merging. Please ignore during review / do not block approval

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:212]: Note: the following are the changes done below
- Extract separate functions for (The logic is kept the same, simply new functions for readability and testability)
  * creating pool denoms map
  * creating filtered balances
  * compute pool liquidity capitalization in uosmo

Added logic:
- Convert uosmo liquidity capitalization to USDC and descale using the precision scaling factor

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [inline ingest/sqs/pools/transformer/pool_transformer.go:128]: Note: The asset list is not needed anymore and can be removed. Now, we only depend on it in `sqs`

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (czarcas7ic) [inline ingest/sqs/pools/transformer/pool_transformer.go:118]: I think this needs renaming, this is no longer a map from denom to poolID given it is a BigDec now

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (czarcas7ic) [inline ingest/sqs/pools/transformer/pool_transformer.go:294]: Is this some historical chain side logic that is no longer possible? I feel like this being possible now would break some important invariants..

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (czarcas7ic) [reply ingest/sqs/pools/transformer/pool_transformer_test.go:37]: ^ agree

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (czarcas7ic) [inline ingest/sqs/pools/transformer/pool_transformer_test.go:80]: I think UOSMOPool not UOSMOCPool

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (czarcas7ic) [inline ingest/sqs/pools/transformer/pool_transformer_test.go:90]: var name change

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (czarcas7ic) [inline ingest/sqs/pools/transformer/pool_transformer_test.go:127]: Why is that?

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (czarcas7ic) [inline ingest/sqs/pools/transformer/pool_transformer_test.go:517]: No longer routing info, along with the test names below that reference it

**Review by czarcas7ic — APPROVED** [review]
Skimmed the test file a bit but the rest looks good!

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (p0mvn) [reply ingest/sqs/pools/transformer/pool_transformer_test.go]: searched and renamed all referenced

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [reply ingest/sqs/pools/transformer/pool_transformer.go:118]: Done

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (p0mvn) [reply ingest/sqs/pools/transformer/pool_transformer.go:294]: It is possible but the logic is equivalent to what used to be before. We still need to filter those out.

I simply extracted this into a separate function for readability and testability

File: ingest/sqs/pools/transformer/pool_transformer_test.go
Comment (p0mvn) [reply ingest/sqs/pools/transformer/pool_transformer_test.go:127]: Added clarifying comment:

```
// This is for testing that the price is picked up from
// this map, if present, rather than pool.
```

File: ingest/sqs/pools/transformer/pool_transformer.go
Comment (PaddyMc) [inline ingest/sqs/pools/transformer/pool_transformer.go:51]: should this be empty?

**Review by PaddyMc — APPROVED** [review]
One nit, but gg besides :mage:

**Comment by p0mvn** [comment]
Blocked on merge of:
- https://github.com/osmosis-labs/sqs/pull/241
- https://github.com/osmosis-labs/sqs/pull/209

We will merge this PR once SQS side is ready

**Pull request merged**
